staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -2605,8 +2605,20 @@ func (c *Cloud) CreateDisk(volumeOptions *VolumeOptions) (KubernetesVolumeID, er
 		// because Kubernetes may have limited permissions to the key.
 		if isAWSErrorVolumeNotFound(err) {
 			err = fmt.Errorf("failed to create encrypted volume: the volume disappeared after creation, most likely due to inaccessible KMS encryption key")
+		} else {
+			// When DescribeVolumes api failed, plugin will lose track on the volumes' state
+			// driver should be able to clean up these kind of volumes to make sure they are not leaked on customers' account
+			awsDisk, error := newAWSDisk(c, volumeName)
+			if error != nil {
+				return KubernetesVolumeID(""), fmt.Errorf("Failed to create volume: the volume is not in the desired state after waiting, and unable to cleanup leaked volume on SP. VolumeName: %v", volumeName)
```

Comment:
I'm not sure if we should return these errors here. I think we still want to return the original error, ie, why the createVolume failed. These errors can be simply logged instead of returned.

Actually now that I think about it, maybe we should have a log "Failed to create the volume due to <error>. Will try to delete it." right after else above. Otherwise it's hard to understand what's happening and why without reading the code.

Reply:
Yeah that's a good point, it is updated. ty